executor/common_kvm_arm64_syzos.h

```diff
@@ -121,3 +132,24 @@ guest_handle_msr(uint64 reg, uint64 val)
 	    : [val] "r"(val), [pc] "r"(insn)
 	    : "x0", "x30", "memory");
 }
+
+// See "SMC Calling Convention", https://documentation-service.arm.com/static/5f8ea482f86e16515cdbe3c6
+GUEST_CODE static void guest_handle_smc(struct api_call_smc* cmd)
+{
+	asm volatile(
+	    "mov x0, %[smc_id]\n"
+	    "mov x1, %[arg1]\n"
+	    "mov x2, %[arg2]\n"
+	    "mov x3, %[arg3]\n"
+	    "mov x4, %[arg4]\n"
+	    "mov x5, %[arg5]\n"
+	    // TODO(glider): it could be interesting to pass other immediate values here, although
+	    // they are ignored as per the calling convention.
+	    "smc #0\n"
+	    : // Ignore the outputs for now
+	    : [smc_id] "r"((uint32)cmd->smc_id),
+	      [arg1] "r"(cmd->params[0]), [arg2] "r"(cmd->params[1]),
+	      [arg3] "r"(cmd->params[2]), [arg4] "r"(cmd->params[3]),
+	      [arg5] "r"(cmd->params[4])
+	    : "x0", "x1", "x2", "x3", "x4", "x5", "memory");
```

Comment:
The doc says "Registers X18-X30 and stack pointers SP_EL0 and SP_ELx are callee-saved and must be preserved over the SMC call".

Should we clobber the rest?
Is there something like "clobbers all general purpose registers"?

Reply:
Wait, but if they are callee-saved, we shouldn't be preserving them in the caller?

Reply:
Not these, the rest of registers (these are presumably caller-saved).

Reply:
You're right, I'll add x6-x17 to the clobber list.